utils/zerovec/src/ule/chars.rs

```diff
@@ -0,0 +1,108 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+//! ULE implementation for the `char` type.
+
+use super::*;
+use std::convert::TryFrom;
+
+/// A u8 array of little-endian data corresponding to a Unicode code point.
+///
+/// The bytes of a `CharULE` are guaranteed to represent a little-endian-encoded u32 that is a
+/// valid `char` and can be converted without validation.
+#[repr(transparent)]
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub struct CharULE([u8; 4]);
+
+impl ULE for CharULE {
+    type Error = std::char::CharTryFromError;
+
+    #[inline(always)]
+    fn parse_byte_slice(bytes: &[u8]) -> Result<&[Self], Self::Error> {
+        // Validate the bytes
+        for chunk in bytes.chunks_exact(4) {
+            // TODO: Use slice::as_chunks() when stabilized
+            let u = u32::from_le_bytes([chunk[0], chunk[1], chunk[2], chunk[3]]);
+            char::try_from(u)?;
+        }
```

Comment:
`request`: can we have an unsafe `_unchecked` for known validated data here and in public `ZeroVec` constructor?

Reply:
I would rather keep these traits with as few methods as possible until we have a clear use case for adding more.

A major value proposition of ZeroVec is that these operations are safe on untrusted data.  We don't have a clear use case right now for unsafely parsing bytes.  If one comes up, we can figure out the best way to do it.